packages/common/src/api/tan-query/useCollectionByPermalink.ts

```diff
@@ -0,0 +1,67 @@
+import { useQuery, useQueryClient } from '@tanstack/react-query'
+import { useDispatch, useSelector } from 'react-redux'
+
+import { userCollectionMetadataFromSDK } from '~/adapters/collection'
+import { useAppContext } from '~/context/appContext'
+import { OptionalId } from '~/models'
+import { getUserId } from '~/store/account/selectors'
+
+import { QUERY_KEYS } from './queryKeys'
+import { primeCollectionData } from './utils/primeCollectionData'
+
+export const playlistPermalinkToHandleAndSlug = (permalink: string) => {
+  const splitPermalink = permalink.split('/')
+  if (splitPermalink.length !== 4) {
+    throw Error(
+      'Permalink formatted incorrectly. Should follow /<handle>/playlist/<slug> format.'
+    )
+  }
+  const [, handle, , slug] = splitPermalink
+  return { handle, slug }
+}
+
+type Config = {
+  staleTime?: number
+  enabled?: boolean
+}
+
+export const useCollectionByPermalink = (
+  permalink: string | undefined | null,
+  config?: Config
+) => {
+  const { audiusSdk } = useAppContext()
+  const queryClient = useQueryClient()
+  const dispatch = useDispatch()
+  const currentUserId = useSelector(getUserId)
+
+  return useQuery({
+    queryKey: [QUERY_KEYS.collectionByPermalink, permalink],
+    queryFn: async () => {
+      if (!permalink) return null
+      const { handle, slug } = playlistPermalinkToHandleAndSlug(permalink)
+      const { data = [] } =
+        await audiusSdk!.full.playlists.getPlaylistByHandleAndSlug({
```

Comment:
noticed this elsewhere in the repo, but i think we should probably avoid the non-null assertion operator as much as possible? probably a check for the existence of audiusSdk with an early return suffices?

Reply:
sure! it can't really happen because the "enabled" flag, but down to avoid as much as possible. we can def add a rule as well